backend/src/packages/orders/order.service.ts

```diff
@@ -301,6 +411,21 @@ class OrderService implements Omit<IService, 'find'> {
       });
     }
   }
+
+  private checkIsOrderAccepted(
+    isAccepted: boolean,
+    user: UserEntityObjectWithGroupT | null,
+  ): OrderStatusValues {
+    if (user && checkIsDriver(user.group.key)) {
+      return isAccepted ? OrderStatus.CONFIRMED : OrderStatus.REFUSED;
+    } else if ((!user || checkIsCustomer(user.group.key)) && !isAccepted) {
+      return OrderStatus.CANCELED;
+    } else {
+      throw new NotFoundError({
+        message: HttpMessage.USER_CAN_NOT_ACCEPT_OR_DECLINE_ORDER,
+      });
+    }
```

Comment:
I believe it can be interpreted like this
```
if (checkIsDriver(user?.group.key ?? null)) {
  return...
}

if (...) {
  return OrderStatus.CANCELED;
}

throw new NotFoundError({})
```